.github/workflows/test.yaml

```diff
@@ -0,0 +1,34 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+name: File Change Warning
+
+on:
+  pull_request:
+    branches: [main]
+    types: [opened, reopened, ready_for_review, synchronize]
+
+# If there is a new commit, the previous jobs will be canceled
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  file-change-detection:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Check for changed Dockerfile paths
+        run: |
+          merged_commit=$(git log -1 --format='%H')
```

Comment:
is hours granularity adequate?

Reply:
`-1`: This option limits the output to just the most recent commit.
`--format='%H'`: This specifies the format of the output. %H is a format specifier that outputs the full commit hash.